### PR TITLE
[new-backend] resolver: add strategy for sudo user:/ namespace

### PR DESCRIPTION
**Pull request by atmaxinger**
I have added a resolver strategy called `s` for the `user:/` namespace.
If the environment variable `SUDO_USER` is set, it will use passwd to resolve the path for this user.
Unforunately, the tests do not work (yet), but this seems to be the case of the `new-backend` branch - not quite sure how to fix them yet.

## Basics

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

File: doc/COMPILE.md [hunk 1]
```diff
@@ -324,7 +324,8 @@ The following base flags are available:
 
 The user flags are (the order matters!):
 
-- `p` use passwd/ldap to lookup home directory using `getpwuid_r`
+- `p` use passwd/ldap to lookup home directory using `getpwnam_r` with the username from the POSIX environment variable LOGNAME
+- `s` same as `p` but with the environment variable SUDO_USER, which enables access to the `user:/` namespace of the executing user when using `sudo`
```
Comment (markus2330) [inline doc/COMPILE.md:328]: Please also support `DOAS_USER`. Please move this documentation to the resolver and link from here to the resolver.

File: doc/COMPILE.md [hunk 2]
```diff
@@ -324,7 +324,8 @@ The following base flags are available:
 
 The user flags are (the order matters!):
 
-- `p` use passwd/ldap to lookup home directory using `getpwuid_r`
+- `p` use passwd/ldap to lookup home directory using `getpwnam_r` with the username from the POSIX environment variable LOGNAME
```
Comment (markus2330) [inline doc/COMPILE.md:327]: Please leave the behavior of this flag unchanged. Using LOGNAME is less reliable.

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great job! Please do not forget to add a line to the release notes. Otherwise the build server does not do anything.

File: doc/COMPILE.md (hunk 1 above)
Comment (atmaxinger) [reply doc/COMPILE.md]: done.

File: src/plugins/resolver/filename.c [hunk 3]
```diff
@@ -138,7 +138,35 @@ static char * elektraResolvePasswd (Key * warningsKey)
...
-	int s = getpwuid_r (getuid (), &pwd, buf, bufSize, &result);
+	int s = getpwuid_r  (getuid (), &pwd, buf, bufSize, &result);
+
+	if (result == NULL)
+	{
+		elektraFree (buf);
+		if (s != 0)
+		{
+			ELEKTRA_ADD_INSTALLATION_WARNINGF (warningsKey, "Could not retrieve from passwd using getpwuid_r. Reason: %s",
+							   strerror (s));
```
Comment (markus2330) [inline src/plugins/resolver/filename.c:149]: Please use strerror_r. Maybe you find also a beautiful way how to use hide this within a macro as it is a very common thing to add warnings/set errors with a string from errno.

**Review by markus2330 — CHANGES_REQUESTED** [review]
I hope it is okay that I review your draft PRs. Ignore the comments if you already knew about it :wink:

**Pull request closed without merging**